src/redux/books/books.js

```diff
@@ -59,4 +19,101 @@ export const removeBook = (id) => ({
   payload: { id },
 });
 
+export const fetchBooks = (books) => ({
+  type: FETCH_BOOKS,
+  payload: books,
+});
+
+export const fetchBooksError = (error) => ({
+  type: FETCH_BOOKS_ERROR,
+  payload: error,
+});
+
+export const fetchBooksLoading = () => ({
+  type: FETCH_BOOKS_LOADING,
+});
+
+// Default state
+const books = {
+  books: [],
+  loading: false,
+  error: null,
+};
+
+export const getBooks = () => (dispatch) => {
+  dispatch(fetchBooksLoading());
+  fetch(URL)
+    .then((response) => response.json())
+    .then((data) => {
+      const newBooks = [];
+      Object.keys(data).forEach((key) => {
+        if (key) {
+          newBooks.push({
+            ...data[key][0],
+            item_id: key,
+          });
+        }
+      });
+      dispatch(fetchBooks(newBooks));
+    })
+    .catch((error) => {
+      dispatch(fetchBooksError(error.message));
+    });
+};
+
+export const postBook = (book) => (dispatch) => {
+  fetch(URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify((book)),
+  }).then(() => {
+    dispatch(addBook(book));
+  });
+};
+
+export const deleteBook = (id) => (dispatch) => {
+  fetch(`${URL}/${id}`, {
+    method: 'DELETE',
+  })
+    .then(() => {
+      dispatch(getBooks());
+    });
```

Comment:
- [ ] Kindly make sure you dispatch an action for removing(`dispatch(removeBook(id))`) the book in the store instead of getting all books from the API(`dispatch(getBooks))`

Reply:
- [ ] As the last code reviewer said you should use removeBook action inside the dispatch. Kindly change it and make sure everything is working after changing it.
Instead of `dispatch(getBooks())` you should use` dispatch(removeBook(id))`

Reply:
Resolved.